python/shotgun_model/shotgun_model.py

```diff
@@ -841,6 +865,62 @@ def _load_external_data(self):
         """
         pass
 
+    def _get_additional_columns(self, item, is_leaf, columns):
+        """
+        Called when an item is about to be inserted into the model, to get additional items
+        to be included in the same row as the specified item. This provides an opportunity
+        for subclasses to create one or more additional columns for each item in the model.
+
+        Note that this method is always called before inserting an item, even when loading
+        from the cache. Any data that is expensive to compute or query should be added
+        to the ShotgunStandardItem in _populate_item, since column data is not cached.
+        Also note that item population methods (_populate_item, _populate_thumbnail, etc)
+        will not be called on the return columns.
+
+        This method should return a list of QStandardItems, one for each additional column.
+        The original ShotgunStandardItem is always the first item in each row and should
+        NOT be included in the returned list. Any empty value returned by this method
+        is guaranteed to be treated as an empty list (i.e. you may return None).
+
+        This method is called after _finalize_item.
+
+        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model
```

Comment:
Minor: Wondering if the code would be clearer if we indicated that this is the `item` representing the first or the main column of the data we are processing. Maybe call it `primary_item`? (since it carried the payload with all the shotgun data etc?